builtin/providers/aws/resource_aws_codepipeline.go

```diff
@@ -0,0 +1,504 @@
+										Required: true,
+									},
+									"version": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"input_artifacts": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"output_artifacts": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"role_arn": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"run_order": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+func validateAwsCodePipelineEncryptionKeyType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"KMS": true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: encryption_key type can only be KMS"))
+	}
+	return
+}
+
+func validateAwsCodePipelineArtifactStoreType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"S3": true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: artifact_store type can only be S3"))
+	}
+	return
+}
+
+func validateAwsCodePipelineStageActionCategory(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"Source":   true,
+		"Build":    true,
+		"Deploy":   true,
+		"Test":     true,
+		"Invoke":   true,
+		"Approval": true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: category can only be one of Source | Build | Deploy | Test | Invoke | Approval"))
+	}
+	return
+}
+
+func validateAwsCodePipelineStageActionOwner(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"AWS":        true,
+		"ThirdParty": true,
+		"Custom":     true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: owner can only be one of AWS | ThirdParty | Custom"))
+	}
+	return
+}
+
+func validateAwsCodePipelineStageActionConfiguration(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(map[string]interface{})
+	types := map[string]bool{
+		"OAuthToken": true,
+	}
+
+	for k := range value {
+		if types[k] {
+			errors = append(errors, fmt.Errorf("CodePipeline: OAuthToken should be set as environment variable 'GITHUB_TOKEN'"))
+		}
+	}
+	return
+}
+
+func resourceAwsCodePipelineCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codepipelineconn
+	pipeline := expandAwsCodePipeline(d)
+	params := &codepipeline.CreatePipelineInput{
+		Pipeline: pipeline,
```

Comment:
I think, as `expandAwsCodePipeline` func doesn't return an error, then we can just do it as follows:

`Pipeline: expandAwsCodePipeline(d)`